sdcm/nemesis.py

```diff
@@ -3820,6 +3820,108 @@ def _shrink_cluster(self, rack=None):
         self.log.info("Cluster shrink finished. Current number of nodes %s", num_of_nodes)
         InfoEvent(message=f'Cluster shrink finished. Current number of nodes {num_of_nodes}').publish()
 
+    # TODO: add support for the 'LocalFileSystemKeyProviderFactory' and 'KmipKeyProviderFactory' key providers
+    # TODO: add encryption for a table with large partitions?
+
+    def disrupt_enable_disable_table_encryption_aws_kms_provider(self):
+        # TODO: add check for Scylla that it supports the feature
+        if self.cluster.params.get("cluster_backend") != "aws":
+            raise UnsupportedNemesis("This nemesis is supported only on the AWS cluster backend")
+        if not self.cluster.params.get("aws_instance_profile_name_db"):
+            raise UnsupportedNemesis("This nemesis is supported only when an instance profile is used")
+        # TODO: update the 'kms_host' parameter when  dynamic key creation gets implemented.
+        self._enable_disable_table_encryption(additional_scylla_encryption_options={
+            'key_provider': 'KmsKeyProviderFactory', 'kms_host': 'kms_test'})
+
+    def _enable_disable_table_encryption(self, additional_scylla_encryption_options=None):
+        if additional_scylla_encryption_options and additional_scylla_encryption_options.get(
+                'key_provider', 'N/A') == 'KmsKeyProviderFactory':
+            for node in self.cluster.nodes:
+                is_restart_needed = False
+                with node.remote_scylla_yaml() as scylla_yml:
+                    if not scylla_yml.kms_hosts:
+                        scylla_yml.kms_hosts = {}
+                    # TODO: update the 'kms_hosts' field when the dynamic key creation gets implemented.
+                    if 'kms_test' not in scylla_yml.kms_hosts:
+                        scylla_yml.kms_hosts['kms_test'] = {
+                            'master_key': 'alias/kms_encryption_test',
+                            'aws_region': 'us-east-1',
+                            'aws_use_ec2_credentials': True,
+                        }
+                        is_restart_needed = True
+                if is_restart_needed:
+                    node.configure_kms()
+                    node.process_scylla_args('--logger-log-level kms=trace')
+                    node.restart_scylla()
+
+        # Create table with encryption
+        keyspace_name, table_name = self.cluster.get_test_keyspaces()[0], 'tmp_encrypted_table'
+        scylla_encryption_options = {'cipher_algorithm': 'AES/ECB/PKCS5Padding', 'secret_key_strength': 128}
+        scylla_encryption_options |= additional_scylla_encryption_options or {}
+        self.log.info("Create '%s.%s' table with server encryption", keyspace_name, table_name)
+        with self.cluster.cql_connection_patient(self.target_node, keyspace=keyspace_name) as session:
+            # NOTE: scylla-bench expects following table structure:
```

Comment:
don't we have a function for pre-create of s-b table ?

Reply:
No, there is only `c-s`-related functionality.